### ISSUE TITLE
Windows MSI embedded pip install operation warnings

### ISSUE THREAD

**Issue by digimaun** [Azure/azure-cli#6686]
**Describe the bug**
Using pip from the Windows MSI CLI environment causes a number of ‘X requires Y which is not installed’ messages. 

Example:
```
knack 0.3.3 requires pyyaml, which is not installed.
knack 0.3.3 requires tabulate, which is not installed.
keyring 13.1.0 requires pywin32-ctypes!=0.1.0,!=0.1.1; sys_platform == "win32", which is not installed.
humanfriendly 4.12.1 requires pyreadline; sys_platform == "win32", which is not installed.
cffi 1.11.5 requires pycparser, which is not installed.
azure-cli-interactive 0.3.23 requires pyyaml~=3.12, which is not installed.
azure-cli-cosmosdb 0.1.21 requires pydocumentdb>=2.0.1, which is not installed.
azure-cli-core 2.0.39 requires pyyaml, which is not installed.
azure-cli-core 2.0.39 requires tabulate<=0.8.2,>=0.7.7, which is not installed.
azure-cli-container 0.1.24 requires pyyaml, which is not installed.
azure-cli-appservice 0.1.35 requires vsts-cd-manager<1.1.0, which is not installed.
azure-cli-acs 2.1.1 requires pyyaml, which is not installed.
azure-cli-acs 2.1.1 requires sshtunnel, which is not installed.
```

**To Reproduce**
Fast way to reproduce....

0) Install using latest Windows MSI (2.0.39 as of this writing)
1) install the IoT extension
2) run 'az iot hub monitor-events' ....this will do an out of band uamqp install in the extension space using the embedded pip.
3) uamqp installs, the command behavior works...but the aforementioned warnings gets output.

**Expected behavior**
pip install operation to work without the extraneous text

**Environment summary**
Windows MSI (2.0.39 as of this writing), previous versions are also affected...I think 2.0.33 was the last version which this was not happening. 

**Additional context**

**Comment by troydai** [comment]
@williexu can you dive into this scenario a little bit? I'm a bit concern about using the pip to alter the dependencies here.

**Comment by digimaun** [comment]
Resolved in v0.5.1 of the IoT extension, wrapping output with subprocess.check_output similar to the extension command module implementation.

This does not resolve underlying elements causing the messages but is a sufficient workaround for my use case.
